test/test/nonjava/common_non_java.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "common_non_java.h"
+
+asprof_init_t CommonNonJava::asprof_init;
+asprof_execute_t CommonNonJava::asprof_execute;
+asprof_error_str_t CommonNonJava::asprof_error_str;
+
+JavaVM* CommonNonJava::jvm;
+JNIEnv* CommonNonJava::env;
+
+void* CommonNonJava::jvmLib;
+
+void CommonNonJava::outputCallback(const char* buffer, size_t size) {
+    fwrite(buffer, sizeof(char), size, stderr);
+}
+
+void CommonNonJava::loadProfiler() {  
+    #ifdef __linux__
+    void* lib = dlopen("build/lib/libasyncProfiler.so", RTLD_NOW | RTLD_GLOBAL);
+    #else
+    void* lib = dlopen("build/lib/libasyncProfiler.dylib", RTLD_NOW | RTLD_GLOBAL);
+    #endif
+    if (lib == NULL) {
+        fprintf(stderr, "%s\n", dlerror());
+        exit(1);
+    }
+
+    asprof_init = (asprof_init_t)dlsym(lib, "asprof_init");
+    asprof_execute = (asprof_execute_t)dlsym(lib, "asprof_execute");
+    asprof_error_str = (asprof_error_str_t)dlsym(lib, "asprof_error_str");
+
+    asprof_init();
+}
+
+void CommonNonJava::startProfiler() {
+    asprof_error_t err = asprof_execute("start,event=cpu,interval=1ms,cstack=dwarf", outputCallback);
+    if (err != NULL) {
+        fprintf(stderr, "%s\n", asprof_error_str(err));
+        exit(1);
+    }
+}
+
+void CommonNonJava::stopProfiler(char* outputFile) {
+    char cmd[4096];
+    snprintf(cmd, sizeof(cmd), "stop,file=%s", outputFile);
+    asprof_error_t err = asprof_execute(cmd, outputCallback);
+    if (err != NULL) {
+        fprintf(stderr, "%s\n", asprof_error_str(err));
+        exit(1);
+    }
+}
+
+void CommonNonJava::loadJvmLib() {
+    char* java_home = getenv("JAVA_HOME");
+    if (java_home == NULL) {
+        fprintf(stderr, "JAVA_HOME is not set\n");
+        exit(1);
+    }
+    char libPath[4096];
+    #ifdef __linux__
```

Comment:
`#ifdef` inside a function again.
Since you use this for choosing between .so/.dylib in multiple places, it may be better to declare a platform-specific constant or a macro at the top of the file.